### DISCUSSION TITLE
RuntimeError: applying transform <monai.transforms.compose.Compose object at 0x7f75b590b550>

### DISCUSSION THREAD

**Discussion by lincong8722** [Project-MONAI/MONAI#4756] [Q&A]
When I use the 3D nii data of dimension (2, 256, 256, 256) (where, 2 is the number of channels, I stitch the two (256, 256, 256) 3d images together by the newly created dimension ) input into the network, the program will report the following error:
![image](https://user-images.githubusercontent.com/57218642/180719704-114ccc20-0cb9-450d-9155-35b593a3810c.png)
![image](https://user-images.githubusercontent.com/57218642/180719766-97bd2465-702c-4c6d-a0c1-9c1635d0f7cb.png)
————————————————————
This is my part of code:
```
train_transforms = Compose([ScaleIntensity(),AsChannelFirst(),
                                Resize((156, 156, 156)),
                                RandSpatialCrop(roi_size=(128, 128, 128),random_size=False),
                                RandFlip(prob=0.3,spatial_axis=1),
                                RandRotate(prob=0.3,range_x=0.2),
                                RandCoarseDropout(prob=0.3,holes=100,spatial_size=20,fill_value=0),
                                EnsureType()])
    val_transforms = Compose([ScaleIntensity(),AsChannelFirst(),Resize((128,128,128)),EnsureType()])

    # Define image dataset, data loader
    check_ds = ImageDataset(image_files=images_pre, labels=labels, transform=train_transforms)
    check_loader = DataLoader(check_ds, batch_size=4, num_workers=2, pin_memory=torch.cuda.is_available())
    im, label = monai.utils.misc.first(check_loader)
```
This should be a problem with my transformer part, but I don't know how to write the transformer if the input nii data itself has a channel?

**Comment by KumoLiu** [comment]
Hi @lincong8722,
It seems you have a problem with the dim of the input data. If your input data has shape with `(2, 256, 256, 256)`, you may not need to use `AsChannelFirst` which will add a channel dim at the first dim. Check the input data shape first.
Hope it will helps you, thanks!
